inflection/src/inflection/dialog/DisplayValue.cpp

```diff
@@ -21,14 +24,26 @@ DisplayValue::DisplayValue(const ::std::u16string& value)
 {
 }
 
-DisplayValue::DisplayValue(const SpeakableString& value, const SemanticFeature& speakFeature)
-    : DisplayValue(value.getPrint(), {})
+DisplayValue::DisplayValue(
+    const SpeakableString& value,
+    const SemanticFeature& speakFeature, 
+    const ::std::map<SemanticFeature, ::std::u16string> constraintMap)
+    : DisplayValue(value.getPrint(), constraintMap)
 {
     if (!value.speakEqualsPrint()) {
-        constraintMap.emplace(speakFeature, value.getSpeak());
+        this->constraintMap.emplace(speakFeature, value.getSpeak());
     }
 }
 
+DisplayValue::DisplayValue(
+    const SpeakableString& value,
+    const SemanticFeature& speakFeature)
+    : DisplayValue(value.getPrint(), {})
+{
+    if (!value.speakEqualsPrint()) {
+        this->constraintMap.emplace(speakFeature, value.getSpeak());
+    }
+}
```

Comment:
Can you please declare these functions in the same order as the header? It makes it easier to navigate.

Also the 2 argument constructor with the speakFeature can now call the new constructor with an empty map. Then the redundant if statement can be removed. It's best to keep the copy and paste code to a minimum to improve the maintenance.